.github/workflows/platform-autgpt-deploy-prod.yml

```diff
@@ -15,6 +15,49 @@ env:
   NAMESPACE: prod-agpt
 
 jobs:
+  migrate:
+    environment: production
+    name: Run migrations for AutoGPT Platform
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install prisma
+
+      - name: Check for changes
+        id: check_changes
+        run: |
+          git fetch origin dev
+          BACKEND_CHANGED=$(git diff --name-only origin/master HEAD | grep "^autogpt_platform/backend/migrations" && echo "true" || echo "false")
+          MARKET_CHANGED=$(git diff --name-only origin/master HEAD | grep "^autogpt_platform/market/migrations" && echo "true" || echo "false")
+          echo "backend_changed=$BACKEND_CHANGED" >> $GITHUB_OUTPUT
+          echo "market_changed=$MARKET_CHANGED" >> $GITHUB_OUTPUT
+
+      - name: Run Backend Migrations
+        if: steps.check_changes.outputs.backend_changed == 'true'
+        working-directory: ./autogpt_platform/backend
+        run: |
+          python -m prisma migrate deploy
+        env:
+          DATABASE_URL: ${{ secrets.BACKEND_DATABASE_URL }}
+
+      - name: Run Market Migrations
+        if: steps.check_changes.outputs.market_changed == 'true'
+        working-directory: ./autogpt_platform/market
+        run: |
+          python -m prisma migrate deploy
+        env:
+          DATABASE_URL: ${{ secrets.MARKET_DATABASE_URL }}
```

Comment:
is there a reason we wouldn't always run migrations?

Reply:
when there are no changes in the migrations?

Reply:
Doesn’t the tool determine that?

Reply:
If I'm crazy feel free to resolve btw

Reply:
fair i was thinking we just don't do this step at all thats why. but ill update!